src/main/java/uk/gov/hmcts/ccd/domain/service/processor/DateTimeEntryProcessor.java

```diff
@@ -0,0 +1,87 @@
+package uk.gov.hmcts.ccd.domain.service.processor;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CaseViewFieldBuilder;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CommonField;
+import uk.gov.hmcts.ccd.domain.types.BaseType;
+import uk.gov.hmcts.ccd.domain.types.CollectionValidator;
+import uk.gov.hmcts.ccd.endpoint.exceptions.DataProcessingException;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static uk.gov.hmcts.ccd.domain.model.definition.FieldType.*;
+import static uk.gov.hmcts.ccd.domain.service.processor.DisplayContextParameter.*;
+
+@Component
+public class DateTimeEntryProcessor extends CaseDataFieldProcessor {
+
+    private static final List<String> SUPPORTED_TYPES = Arrays.asList(DATETIME, DATE);
+
+    private final DateTimeFormatParser dateTimeFormatParser;
+
+    @Autowired
+    public DateTimeEntryProcessor(CaseViewFieldBuilder caseViewFieldBuilder,
+                                  DateTimeFormatParser dateTimeFormatParser) {
+        super(caseViewFieldBuilder);
+        this.dateTimeFormatParser = dateTimeFormatParser;
+    }
+
+    @Override
+    protected JsonNode executeSimple(JsonNode node, CommonField field, BaseType baseType, String fieldPath) {
+        return !isNullOrEmpty(node)
+            && hasDisplayContextParameterType(field.getDisplayContextParameter(), DisplayContextParameterType.DATETIMEENTRY)
```

Comment:
To improve code readability and reduce duplication is essential to introduce utility method in the domain classes where they belong to. 
`hasDisplayContextParameterType` should be removed. And instead we should introduced  a method in `DisplayContextParameter` class that allows to check the type of a dcp. For example:
`displayContextParameter.hasType(DisplayContextParameterType.DATETIMEENTRY)`
Then we could add a method to CommonField that returns the display context parameter as an instance of `DisplayContextParameter` other than a String. 
If we do that this will simplify in:
`field.getDisplayContextParameter.hasType(DisplayContextParameterType.DATETIMEENTRY)` 
This is much more reusable and will help in reducing duplication